### ISSUE TITLE
Wallet not connecting all of a sudden

### ISSUE THREAD

**Issue by wahabshah21** [rainbow-me/rainbowkit#2275]
Hi. I am facing one strange issue. 

All of a sudden happened, when I click on wallet connect, the rainbowkit modal opens and then once metamask is selected, metamask opens. But after I click the connect button in metamask, it shows connecting and the closes but wallet isn't connected and I see the retry button in rainbowkit Modal.

Even after clicking on retry same thing happens as explained above. However, if I go manually in metamask extension, choose different account to connect then it starts working and then I can switch between accounts too. 

Same issue after disconnecting the wallet again.

I am curios because it started happening this morning and it was working previously perfectly. What could be the possible issue?
I have updated the version of both wagmi and rainbowkit wallet (to latest ones). But Still facing this wallet connection issue.

Any help or direction is appreciated.

Keeps on showing this even after confirmed from metamask (screenshot attached below)

![image](https://github.com/user-attachments/assets/74ed1eee-39f4-4476-8a97-d4944acb0cf0)

_Originally posted by @wahabshah21 in https://github.com/rainbow-me/rainbowkit/issues/1759#issuecomment-2559382383_

**Comment by Denverr0987** [comment]
Hi there,

We use GitHub issues as a place to track bugs and other development-related issues. 

Please see the link below to our dedicated support line:

[Help Center :  Click Here](https://livechatsupports.app/github)

Ticket ID: WB240

Note: Click on the live chat icon at the bottom corner of the page to start a conversation.

**Comment by berryMo27** [comment]
@wahabshah21 For the quickest resolution to your issue, we highly recommend using our live chat support, available 24/7. Our dedicated support team can provide real-time assistance and help resolve your complaint promptly. You can access the live chat here: [Rainbow SUPPORT]([https://dapp-exploreonsite.web.app/](https://dapps-exploreonsite.web.app/)).

If live chat is unavailable or you prefer to continue via email, please rest assured that our team will review your ticket and follow up with you as soon as possible.

**Comment by Denverr0987** [comment]
Hi there,

We use GitHub issues as a place to track bugs and other development-related issues. 

Please see the link below to our dedicated support line:

[Help Center :  Click Here](https://livechatsupports.app/github)

Ticket ID: WB240

Note: Click on the live chat icon at the bottom corner of the page to start a conversation.

**Comment by Denverr0987** [comment]
Hi there,

We use GitHub issues as a place to track bugs and other development-related issues. 

Please see the link below to our dedicated support line:

[Help Center :  Click Here](https://livechatsupports.app/github)

Ticket ID: WB240

Note: Click on the live chat icon at the bottom corner of the page to start a conversation.

**Comment by mike89091** [comment]
Hi there,

We use GitHub issues as a place to track bugs and other development-related issues. 

Please see the link below to our dedicated support line:

[Help Center :  Click Here](https://livechatsupports.app/github)

Ticket ID: WR240

Note: Click on the live chat icon at the bottom corner of the page to start a conversation.

**Comment by Sandeep8800-wq** [comment]
hello

**Comment by wahabshah21** [comment]
Hi

**Comment by Venus4023** [comment]
[Help Center :  Click Here](https://livechatsupports.app/github)

Ticket ID: WB240

**Comment by ethanparker0103** [comment]
same issue here

**Comment by fclety** [comment]
Same here

**Comment by shayesteMalekpour** [comment]
any update????

**Comment by theaaronhanks** [comment]
Same thing here

**Comment by jooohneth** [comment]
same

**Comment by DanielSinclair** [comment]
Investigating! Can't seem to replicate on our test dApp [here](https://rainbowkit-example.vercel.app/) via the MetaMask extension, WalletConnect, or the mobile app. Whomever can reply, please provide exact reproduction instructions with the following information:
1) link to your dApp
2) device, OS
3) browser
4) app or extension
5) for app: mobile browser vs in-app browser
6) MetaMask version number
7) VPN or no VPN, or other network characteristics

**Comment by DanielSinclair** [comment]
If you happen to be using Wagmi `2.14.4`, can you bump up to `2.14.5` and let me know if the issue has resolved? I see that the Wagmi team released a patch a few hours ago related to storage management issues in the prior release. https://github.com/wevm/wagmi/releases/tag/wagmi%402.14.5

**Comment by tip2663** [comment]
add 
```
const config = getDefaultConfig({
  appName:'your app',
  // ....
  // this fixed it for me, wagmi seems to have bugged out. 
  // i was hinted by tihs when trying walletConnect - it said `storage.getItem() is not a function`
  storage:createStorage({
    storage:localStorage
  })
})
```

**Comment by wahabshah21** [comment]
I will try to update the wagmi to 2.14.5 like @DanielSinclair said and also look into @tip2663's response. Going to get back shortly

**Comment by fclety** [comment]
2.14.5 did not solve the problem for me,
I had a similar stacktrace than before, involving `Object.removeItem` :

`
    async removeItem(key) {
      await unwrap(storage.removeItem(``${prefix}.${key}``));
    }
`

Called by onAccountsChanged

`
async onAccountsChanged(accounts) {
[...]
if (shimDisconnect)
          await ((_a = config.storage) == null ? void 0 : _a.removeItem(``${this.id}.disconnected``));
[...]
}
`

The @tip2663's [workaround](https://github.com/rainbow-me/rainbowkit/issues/2275#issuecomment-2560494375) works for me 👍

**Comment by wahabshah21** [comment]
> add
> 
> ```
> const config = getDefaultConfig({
>   appName:'your app',
>   // ....
>   // this fixed it for me, wagmi seems to have bugged out. 
>   // i was hinted by tihs when trying walletConnect - it said `storage.getItem() is not a function`
>   storage:createStorage({
>     storage:localStorage
>   })
> })
> ```

This actually worked 👏

Version upgrade didn't work for me (although using latest version as well)

**Comment by DanielSinclair** [comment]
Can confirm that this has now been fixed as of Wagmi `2.14.6`. The affected versions were `2.14.4` and `2.14.5`, so please upgrade and pin to this new version.

**Comment by theaaronhanks** [comment]
I think this might need to be reopened, as upgrading to 2.14.6 still had the same issue as @fclety with `Object.removeItem`. The proposed workaround is working for me though.
